src/marshmallow/validate.py

```diff
@@ -594,3 +594,47 @@ def __call__(self, value: typing.Sequence[_T]) -> typing.Sequence[_T]:
             if val in self.iterable:
                 raise ValidationError(self._format_error(value))
         return value
+
+
+class NoDuplicates(Validator):
+    """Validator which succeeds if the ``value`` is an ``iterable`` and has no duplicate
+    elements. In case of a list of objects, it can easy check an internal
+    attribute by passing the ``attribute`` parameter.
+    Validator which fails if ``value`` is not a member of ``iterable``.
+
+    :param str attribute: The name of the attribute of the object you want to check.
+    """
+
+    default_message = "Invalid input. Supported lists or str."
+    error = "Found a duplicate value: {value}."
+    attribute_error = "Found a duplicate object attribute ({attribute}): {value}."
+
+    def __init__(self, attribute: typing.Optional[str] = None):
+        self.attribute = attribute
+
+    def _repr_args(self) -> str:
+        return "attribute={!r}".format(self.attribute)
+
+    def _format_error(self, value) -> str:
+        if self.attribute:
+            return self.attribute_error.format(attribute=self.attribute, value=value)
+        return self.error.format(value=value)
+
+    def __call__(self, value):
+        set_item = set()
+        try:
+            for item in value:
+                if self.attribute:
+                    attribute = getattr(item, self.attribute)
```

Comment:
I wonder if we should use `utils.get_value()` here. That would allow validating data buried in deeply nested items using dotted attribute notation.